sympy/functions/elementary/hyperbolic.py

```diff
@@ -473,14 +498,38 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi, cos, sinh
+        arg = self.args[0]
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+
+        # if denom = 0, tanh(arg) = zoo
+        denom = cos(im)**2 + sinh(re)**2
```

Comment:
It is not necessary to construct denom. It is zero if and only if both cos(im) and sinh(re) are zero. This is equivalent to  im % pi == pi/2  and re == 0.